### ISSUE TITLE
Allow runtime-wide `TryState` checks

### ISSUE THREAD

**Issue by ggwpez** [paritytech/substrate#13061]
Currently the `TryState` can only be utilized on a per-pallet level by implementing the `try_state` hook.   
This limits the usage tor multi-pallet `try_state` functionality. `Executive` could be extended to accept an optional `TryState` which will then be executed by the try-runtime CLI on a runtime-wide level.

cc @kianenigma

**Comment by kianenigma** [comment]
Yup, should be an easy one relatively (cc-ing @gpestana @Ank4n).

**Comment by bkchr** [comment]
> `Executive` could be extended to accept an optional `TryState` which will then be executed by the try-runtime CLI on a runtime-wide level.

We don't really need to forward this into `Executive`. We have all pallets as tuple in the runtime and just calling `AllPalletsWithSystem::try_state` should be enough. Before that we would need to change: 
https://github.com/paritytech/substrate/blob/05d11c213b8e097a5368a887b6d055db61ce05db/frame/support/procedural/src/pallet/expand/hooks.rs#L210-L226

to:

```
		#frame_support::try_runtime_enabled! {
			impl<#type_impl_gen>
				#frame_support::traits::TryState<<T as #frame_system::Config>::BlockNumber>
				for #pallet_ident<#type_use_gen> #where_clause
			{
				fn try_state(
					n: <T as #frame_system::Config>::BlockNumber,
					_s: #frame_support::traits::TryStateSelect
				) -> Result<(), &'static str> {
					#log_try_state
					<
						Self as #frame_support::traits::Hooks<
							<T as #frame_system::Config>::BlockNumber
						>
					>::try_state(n)
				}
			}
		}
```

**Comment by kianenigma** [comment]
@bkchr I am not sure why you are against this. The main goal is to enable a try-state check that will look into multiple pallets. The only way to achieve this would be through doing it at the runtime (probably via `executive`)

**Comment by liamaharon** [comment]
So we can already write and execute `TryState` checks in `Executive`? Then we can close this issue?
